src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -80,6 +88,162 @@ private PackageArchiveReader BuildPackage(string path)
+                }
+                return true;
+            }
+
+            return false;
+        }
+
+        private static void LoadProjectJsonFile(PackageBuilder builder, string path, string basePath, string id, Stream stream, Func<string, string> propertyProvider)
+        {
+            string content = Preprocessor.Process(stream, propName => propertyProvider(propName));
+
+            PackageSpec spec = JsonPackageSpecReader.GetPackageSpec(content, id, path);
+
+            if (id == null)
+            {
+                if (basePath.EndsWith(Path.DirectorySeparatorChar.ToString()))
+                {
+                    basePath = basePath.Substring(0, basePath.Length - 2);
+                }
+                builder.Id = Path.GetFileName(basePath);
+            }
+            else
+            {
+                builder.Id = id;
+            }
+            builder.Version = spec.Version;
+            builder.Title = spec.Title;
+            builder.Description = spec.Description;
+            builder.Copyright = spec.Copyright;
+            if (spec.Authors.Any())
+            {
+                builder.Authors.Add(spec.Authors.First());
+            }
+            if (spec.Owners.Any())
+            {
+                builder.Owners.AddRange(spec.Owners);
+            }
+            Uri tempUri;
+            if (Uri.TryCreate(spec.LicenseUrl, UriKind.Absolute, out tempUri))
+            {
+                builder.LicenseUrl = tempUri;
+            }
+            if (Uri.TryCreate(spec.ProjectUrl, UriKind.Absolute, out tempUri))
+            {
+                builder.ProjectUrl = tempUri;
+            }
+            if (Uri.TryCreate(spec.IconUrl, UriKind.Absolute, out tempUri))
+            {
+                builder.IconUrl = tempUri;
+            }
+            builder.RequireLicenseAcceptance = spec.RequireLicenseAcceptance;
+            builder.Summary = spec.Summary;
+            builder.ReleaseNotes = spec.ReleaseNotes;
+            builder.Language = spec.Language;
+
+            // If there's no base path then ignore the files node
+            // Also, id is null only when we want to skip the AddFiles
+            if (basePath != null && id != null)
+            {
+                builder.AddFiles(basePath, @"**\*", null);
+            }
+
+            if (spec.Tags.Any())
+            {
+                builder.Tags.AddRange(spec.Tags);
+            }
+            if (spec.Dependencies.Any())
+            {
+                List<PackageDependency> dependencies = new List<PackageDependency>();
+                foreach (var dependency in spec.Dependencies)
+                {
+                    var flags = dependency.IncludeType.ToString();
+                    
+                    dependencies.Add(new PackageDependency(dependency.Name, dependency.LibraryRange.VersionRange, flags.Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries), new List<string>()));
```

Comment:
This should also check suppressParent. The suppressParent flags are applied for the consuming project, which in this case is the project that installs the nupkg, so basically the include flags for the nupkg dependency within a nupkg is always: include - exclude - suppressParent.

https://github.com/NuGet/Home/wiki/%5BSpec%5D-Managing-dependency-package-assets

If the include flags are empty (or suppressParent is all) then the dependency should not be added to the nupkg at all. These are called type: build dependencies and they are only used by the project itself, not by any consumers.